internal/db/db.go

```diff
@@ -143,115 +144,253 @@ func (tx *holdTx) FindOne(modelID uuid.UUID, matcher Matcher) (rec Record, err e
 	return
 }
 
-func (tx holdTx) FindMany(modelID uuid.UUID, matcher Matcher) []Record {
-	mi := tx.h.IterMatches(modelID, matcher)
-	hits := []Record{}
-	for val, ok := mi.Next(); ok; val, ok = mi.Next() {
-		hits = append(hits, val)
+func (tx *holdTx) FindMany(modelID uuid.UUID, matcher Matcher) (recs []Record, err error) {
+	recs, err = tx.h.FindMany(modelID, matcher)
+	return
+}
+
+func (tx *holdTx) Insert(rec Record) error {
+	tx.ensureWrite()
+	h, err := tx.h.Insert(rec)
+	if err != nil {
+		return err
 	}
-	return hits
+	tx.h = h
+	return nil
 }
 
-func (tx *holdTx) Insert(rec Record) {
+func (tx *holdTx) Update(r Record, fields map[string]interface{}) (Record, error) {
 	tx.ensureWrite()
-	tx.h = tx.h.Insert(rec)
+	//Need to load a model for the record
+	m, _ := tx.GetModel(r.Type())
+	newRec := r.DeepCopy(m)
+	for key, val := range fields {
+		err := newRec.Set(key, val)
+		if err != nil {
+			return nil, err
+		}
+	}
+	h, err := tx.h.Insert(newRec)
+	if err != nil {
+		return nil, err
+	}
+	tx.h = h
+	return newRec, nil
 }
 
-func (tx *holdTx) Connect(left, right Record, rel Relationship) {
+func (tx *holdTx) Connect(left, right Record, rel Relationship) error {
 	tx.ensureWrite()
 
 	if rel.LeftBinding == BelongsTo && (rel.RightBinding == HasOne || rel.RightBinding == HasMany) {
 		// FK left
-		left.SetFK(rel.LeftName, right.ID())
+		id, err := right.ID()
+		if err != nil {
+			return err
+		}
+		err = left.SetFK(rel.LeftName, id)
+		if err != nil {
+			return err
+		}
 	} else if rel.RightBinding == BelongsTo && (rel.LeftBinding == HasOne || rel.LeftBinding == HasMany) {
 		// FK right
-		right.SetFK(rel.RightName, left.ID())
+		id, err := left.ID()
```

Comment:
I wonder if it makes sense to panic on an error for ID. 

If a record doesn't have an ID field, something has gone truly awry.